### ISSUE TITLE
More complex better testing for common problems

### ISSUE THREAD

**Issue by cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw** [huggingface/diffusers#4550]
**What API design would you like to have changed or added to the library? Why?**
Create a better automated testing for pipelines.

by complex i mean It should consist of checking
- if image output,check error has error or not.check output image is it all black?save a few image in case to debug can prevent latent image happening and no one know.
- if audio,check not silent,not constant broken aaaaaaaaaaaaaaaaaa-----

by better it would be in general higher code coverage.
like for example https://github.com/huggingface/diffusers/issues/4520
if there is a standardsdnolorascale_testfunction() and a standardsdwithlorascale_testfunction() 
we would know that standardsdnolorascale_testfunction()  runs perfectly and standardsdwithlorascale_testfunction()  throws error which leads to issue https://github.com/huggingface/diffusers/issues/4520

and honestly its not hard just tedious and more cost to maintain the test,
even a basic python try:catch:finally can be used to solve this issue.

there are 2 way this can be implemented if its not obvious.
1. per function test.possibly many code but yeah .It will perfectly follow diffusers motto least abstraction,but very tedious& redundant(duh abstraction is invented to prevent redundancy) so for each function sometimes you need to create a lot of extra boilercode  like for example this issue https://github.com/huggingface/diffusers/issues/4520 it will NOT get caught if you just test the calling of the function,as the actual error occurs when generation which needs a lot of extra setup.

2. big chunk legit full pipeline test.just go from the start to end with as many custom safetensor,huggingfacehub model, lora,TI EVERYTHING in just a few scripts aka like an app that uses all functionality and you can just see the report and output. open the report open the images ,audio spent a few minute go through pictures and audios weekly,Done.

To prevent common bugs happening in pipelines undetected until community reports

**What use case would this enable or better enable? Can you give us a code example?**
If implemented it will catch these problems in the test before reaching community>
https://github.com/huggingface/diffusers/issues/4520
https://github.com/huggingface/diffusers/issues/4446
https://github.com/huggingface/diffusers/issues/4504

also in most case there is no need for user to even add a reproducible code*

this is just my two cents for improving.as far as I know this is the most efficient way to prevent a lot of future problems as things scales and grows from my experience.

of course there are a lot of ways to prevent bugs but these two are most possible to be used in diffusers

**Comment by sayakpaul** [comment]
Thanks for taking the time to provide your suggestions. 

It's exactly the other way around for us!

We try to not to be very exhaustive in these things and move with the community. This is why, when we release things, we make sure to include the fundamental tests as well as tests that we have learned from our experiences but without clogging the release. 

One example I would give is the unloading functionality of LoRA. We added that after community discovery. As maintainers, we are involved in different things -- feature contribution, training, community engagements, etc. So, this is why, we always rely on community feedback and take necessary steps accordingly.
